Sources/NIOCore/ByteBuffer-aux.swift

```diff
@@ -675,6 +676,23 @@ extension ByteBuffer {
     }
 }
 
+extension ByteBuffer: Codable {
+
+    /// Creates a ByteByffer by decoding from a Base64 encoded single value container.
+    public init(from decoder: Decoder) throws {
+        let container = try! decoder.singleValueContainer()
+        let base64String = try! container.decode(String.self)
+        self = try! ByteBuffer(bytes: base64String.base64Decoded())
```

Comment:
we can't `try!` here, the string might not be valid base64. And we should add a test that proves that.

Reply:
Done: https://github.com/apple/swift-nio/pull/2509/commits/3861cf8b76686fa907d44fb96bf6565483bc5185